compiler/modules/nback/types.bal

```diff
@@ -17,13 +17,25 @@ final llvm:StructType llStructureDescType = llvm:structType([LLVM_TID]);
 // This is an approximation, but close enough since we are only accessing the pointer in C.
 final llvm:StructType llComplexType = llvm:structType([LLVM_BITSET, LLVM_BITSET, llvm:arrayType(llvm:pointerType("i8"), 0)]);
 
-final llvm:FunctionType llListSetFuncType = llvm:functionType(LLVM_PANIC_CODE, [LLVM_TAGGED_PTR, LLVM_INT, LLVM_TAGGED_PTR]);
-final llvm:FunctionType llListGetFuncType = llvm:functionType(LLVM_TAGGED_PTR, [LLVM_TAGGED_PTR, LLVM_INT]);
-final llvm:StructType llListDescType = llvm:structType([LLVM_TID, llvm:pointerType(llListGetFuncType), llvm:pointerType(llListSetFuncType), LLVM_MEMBER_TYPE]);
-final llvm:Type llListType = llvm:structType([llvm:pointerType(llListDescType),                      // *desc
-                                              LLVM_INT,                                              // length
-                                              LLVM_INT,                                              // capacity
-                                              heapPointerType(llvm:arrayType(LLVM_TAGGED_PTR, 0))]); // *members
+final llvm:FunctionType[6] llListDescFuncs = [
+            llvm:functionType(LLVM_TAGGED_PTR, [LLVM_TAGGED_PTR, LLVM_INT]),
+            llvm:functionType(LLVM_PANIC_CODE, [LLVM_TAGGED_PTR, LLVM_INT, LLVM_TAGGED_PTR]),
+            llvm:functionType(LLVM_INT, [LLVM_TAGGED_PTR, LLVM_INT]),
+            llvm:functionType(LLVM_PANIC_CODE, [LLVM_TAGGED_PTR, LLVM_INT, LLVM_INT]),
+            llvm:functionType(LLVM_DOUBLE, [LLVM_TAGGED_PTR, LLVM_INT]),
+            llvm:functionType(LLVM_PANIC_CODE, [LLVM_TAGGED_PTR, LLVM_INT, LLVM_DOUBLE])];
+final llvm:StructType llListDescType = llvm:structType([LLVM_TID,
+                                                        llvm:pointerType(llListDescFuncs[0]),
```

Comment:
Clearer to define a const for each index.

Reply:
Not sure I understand this suggestion. I need `llListDescFuncs` as an array for creating FunctionDesc in a loop.
I can change this to `llvm:pointerType(llListDescFuncs[listGenericArrayRepr.listDescGetIndex -  1])` and so on.

Not sure how I can avoid `llvm:pointerType`.

Reply:
I'll add a function.

Reply:
I tried following, unfortunately it's a JBUG.

```ballerina
function llListDescFuncPtr(int listReprIndex) returns llvm:PointerType {
    return llvm:pointerType(llListDescFuncs[listReprIndex - 1]); // `-1` since TID occupies the first
}

final llvm:StructType llListDescType = llvm:structType([LLVM_TID,
                                                        llListDescFuncPtr(listGenericArrayRepr.listDescGetIndex),
                                                        llListDescFuncPtr(listGenericArrayRepr.listDescSetIndex),
                                                        llListDescFuncPtr(listIntArrayRepr.listDescGetIndex),
                                                        llListDescFuncPtr(listIntArrayRepr.listDescSetIndex),
                                                        llListDescFuncPtr(listFloatArrayRepr.listDescGetIndex),
                                                        llListDescFuncPtr(listFloatArrayRepr.listDescSetIndex),
                                                        LLVM_MEMBER_TYPE])
```

Reply:
Use a loop to create an array of PointerType and initialize the array members in structType from that. You're supposed to be able to use `...x` to splice in the members of the array, but IIRC we haven't got that yet.

Reply:
Point is there is one call to pointerType in the source not six.